lib/elixir/lib/enum.ex

```diff
@@ -1413,15 +1413,18 @@ defmodule Enum do
       6
 
   """
+  @spec sum(Range.t) :: integer
```

Comment:
thank you @lexmag, I noticed of that right after I posted my question.

then, my question is... shouldn't Elixir generate a warning when there are overlapping specs then? or it is something that we shouldn't worry about since it's valid Elixir code and just Dialyzer doesn't like it?